lib/parser/interface-list-parser.ts

```diff
@@ -126,9 +128,65 @@ export class InterfaceListParser extends ParserBase {
 
     }
     if (foundElements === 0) {
-      throw new ParserError(`Empty interface list is not allowed`, startToken.range.copyWithNewEnd(this.getToken().range));
+      let startOffset = 0;
+      while (this.getToken(startOffset, true) !== startToken) {
+        startOffset--;
+      }
+
+      if (this.getToken(startOffset - 1, true).getLText() === 'port'
+      || this.getToken(startOffset - 1, true).getLText() === 'generic'
+      || this.getToken(startOffset - 1, true).getLText() === 'parameter') {
+        startOffset--;
+      }
+      const closingBracket = this.getToken(-1, true);
+      let endDel = closingBracket;
+      if (this.getToken().getLText() === ';') {
+        endDel = this.getToken();
+      }
+      throw new ParserError(`Empty interface list is not allowed`,
+        startToken.range.copyWithNewEnd(closingBracket.range),
+        {
+          message: 'remove interface list',
+          edits: [
+            TextEdit.del(this.getToken(startOffset, true).range.copyWithNewEnd(endDel.range))
+          ]
+        }
+      );
```

Comment:
when merging main (i.e. #165) this should be a diagnostic and not an error because it is recoverable.